src/tiled/changetileprobability.cpp

```diff
@@ -41,10 +41,14 @@ ChangeTileProbability::ChangeTileProbability(MapDocument *mapDocument,
 
 void ChangeTileProbability::swap()
 {
-    float probability = mTile->probability();
-    mTile->setProbability(mProbability);
-    mProbability = probability;
-    mMapDocument->emitTileProbabilityChanged(mTile);
+    QList<float>::iterator itProbability = mProbabilities.begin();
+    for (Tile* tile : mTiles) {
+        float probability = tile->probability();
+        tile->setProbability(*itProbability);
+        *itProbability = probability;
+        mMapDocument->emitTileProbabilityChanged(tile);
+        ++ itProbability;
```

Comment:
I think I'd find this loop more readable as an index-based loop that indexes both arrays, rather than combining a range-based for loop with an iterator.